spec/install/gemfile/git_spec.rb

```diff
@@ -943,6 +943,47 @@
       expect(out).to include("An error occurred while installing foo (1.0)")
       expect(out).not_to include("gem install foo")
     end
+
+    it "does not reinstall the extension" do
```

Comment:
@indirect this whole PR will be good to go once this is scoped to RubyGems > 2.3, I think